sql-db/hibernate-reactive/src/main/java/io/quarkus/ts/hibernate/reactive/http/GroundedEndpoint.java

```diff
@@ -111,7 +111,8 @@ public Uni<Author> createBookThroughSession(String authorName, String name) {
                             Book book = new Book();
                             book.setAuthor(author.getId());
                             book.setTitle(name);
-                            return session3.persist(book);
+                            author.getBooks().add(book);
+                            return session3.persist(author);
                         }))).onItem()
                         .transformToUni(ignore -> factory.withSession(session3 -> session3.find(Author.class, author.getId())))
                         .onFailure()
```

Comment:
On failure there is `Response.Status.BAD_REQUEST` returned, but I would expect `INTERNAL_SERVER_ERROR`.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.1 says
```
10.4.1 400 Bad Request
The request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications.
```

Reply:
Makes sense.